generators/E906LegacyGen/SQPileupGen.C

```diff
@@ -0,0 +1,195 @@
+    PHCompositeNode* dstNode = dynamic_cast<PHCompositeNode*>(iter.findFirst("PHCompositeNode", "DST"));
+
+    _evt = new SQEvent_v1();
+    dstNode->addNode(new PHIODataNode<PHObject>(_evt, "SQEvent", "PHObject"));
+  }
+
+  _mcevt = findNode::getClass<SQMCEvent>(topNode, "SQMCEvent");
+  if(!_mcevt) 
+    {
+      PHNodeIterator iter(topNode);
+      PHCompositeNode* dstNode = dynamic_cast<PHCompositeNode*>(iter.findFirst("PHCompositeNode", "DST"));
+
+      _mcevt = new SQMCEvent_v1();
+      dstNode->addNode(new PHIODataNode<PHObject>(_mcevt, "SQMCEvent", "PHObject"));
+    }
+
+  _extFile = TFile::Open(_extFileName.Data());
+  if(!_extFile)
+    {
+      std::cout << "SQPileupGen: cannot find the input ROOT file, abort " << _extFileName << std::endl;
+      exit(EXIT_FAILURE);
+    }
+  _extTree = (TTree*)_extFile->Get("bkg");
+
+  _extPos = new TClonesArray("TVector3");
+  _extMom = new TClonesArray("TVector3");
+
+  _extTree->SetBranchAddress("eventID", &_extEvtID);
+  _extTree->SetBranchAddress("n", &_nExtPar);
+  _extTree->SetBranchAddress("pdg", _extPDG);
+  _extTree->SetBranchAddress("pos", &_extPos);
+  _extTree->SetBranchAddress("mom", &_extMom);
+
+  _readIdx = 0;
+
+  _vertexGen->InitRun(topNode);
+  return Fun4AllReturnCodes::EVENT_OK;
+}
+
+int SQPileupGen::process_event(PHCompositeNode* topNode)
+{
+  if(_beam_intensity_profile) 
+    {
+      setBucketSize((int)_beam_intensity_profile->GetRandom()/_proton_coeff);//if the function describes dN/dI vs I   
+    }
+
+  if(_beam_intensity_profile_histo)
+    {
+      double QIEcount = _beam_intensity_profile_histo->GetRandom();
+      if (QIEcount>_inhibit_threshold) return Fun4AllReturnCodes::ABORTEVENT;
+      int  bucket_size = (int)QIEcount/_proton_coeff;
+      setBucketSize(bucket_size);       
+    }
+
+  //if(!readExtTree(_bucketSize)) return Fun4AllReturnCodes::ABORTEVENT;
+  if(!readExtTree(_bucketSize)) return Fun4AllReturnCodes::ABORTRUN;
+
+  int eventID_curr = -1;
+  int trackID = 0;
+  TVector3 priVtx;
+  for(auto par = _extParticles.begin(); par != _extParticles.end(); ++par)
+    {
+      if(par->_evtID != eventID_curr)
+	{
+	  eventID_curr = par->_evtID;
+	}
+
+      int vtxidx = _ineve->AddVtx(par->_pos.X(), par->_pos.Y(), par->_pos.Z(), 0.);
+
+      PHG4Particle* thisPar = new PHG4Particlev2();
+      thisPar->set_track_id(trackID++);
+      thisPar->set_vtx_id(vtxidx);
+      thisPar->set_pid(par->_pdg);
+      thisPar->set_px(par->_mom.X());
+      thisPar->set_py(par->_mom.Y());
+      thisPar->set_pz(par->_mom.Z());
+      _ineve->AddParticle(vtxidx, thisPar);
+    }
+
+
+
+  //std::cout<<"bucket size in pileup generator: "<<_bucketSize<<std::endl;
+  _mcevt->set_cross_section(_bucketSize);
+
+  _n_proc_evt++;
+  _evt->set_run_id(0);
+  _evt->set_spill_id(0);
+  _evt->set_event_id(_n_proc_evt);
+  return Fun4AllReturnCodes::EVENT_OK; 
+}
+
+bool SQPileupGen::readExtTree(int nEvents)
+{
+  std::cout<<"nEvents: "<<nEvents<<std::endl;
```

Comment:
Please add `if (Verbosity() > 0)` for example.

Reply:
Thank you. I will follow the suggestion.